src/hooks/useYValue.tsx

```diff
@@ -0,0 +1,127 @@
+import { useCollaboration } from './useCollaboration'
+import { useEffect, useMemo, useState } from 'react'
+import * as Y from 'yjs'
+import { isYArray, isYMap, isYValue } from '@/lib/isType'
+
+type PathArray = Array<string | number>
+type YParent = Y.Array<unknown> | Y.Map<unknown> | undefined
+
+/**
+ * Observe a value in a an exact path. If "deep" is passed, observe all
+ * changes beneath the exact path. This hook is especially usefull when observing (any)
+ * change to a Y.XmlText structure (i.e Textbit/Slate)
+ *
+ * @param path string
+ * @param deep boolean
+ *
+ * @returns [<T>, YParent]
+ */
+export function useYValue<T>(path: string, deep: boolean = false): [
+  T | undefined,
+  YParent
+] {
+  const { provider, synced } = useCollaboration()
+  const yRoot = synced ? provider?.document.getMap('ele') : undefined
```

Comment:
There was something about synced... It only allows server changes, not offline...?